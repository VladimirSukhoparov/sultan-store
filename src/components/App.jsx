import React, { useEffect,useState } from 'react'; 
import { Route, Routes } from "react-router-dom";

import data from './data.json';

import { Header } from './Header';
import { Footer } from './Footer';
import { Breadcrumb } from './Breadcrumb';
import { Home } from './Home';
import { Product } from './Product';
import { Catalog } from './Catalog';
import { Cosmetics } from './Cosmetics';
import { Basket } from './Basket';
import { Admin } from './Admin';

import "./index.scss"

export const App = () => {

  const [basket, setBasket] = useState(JSON.parse(localStorage.getItem('basket')) || []);
  const [products, setProducts] = useState(data);
  
  const manufacturer = Object.values(data.reduce((arr, e) => {
    let manufacturer = e.manufacturer;
    if(!arr[manufacturer]) arr[manufacturer] = {manufacturer, count: 1}
      else arr[manufacturer].count ++;
    return arr;
  }, {}))


  

  return (
    <div className='app'>
      <Header basket={basket} products={products} data={data} setProducts={setProducts}/>
        <div className='content'>
          <Breadcrumb />
            <Routes className='container'>
              <Route path='/' exact element={<Home/>} />
              <Route path='/catalog'  element={<Catalog />} />
              <Route path='/catalog/cosmetics' element={<Cosmetics data={products} setBasket={setBasket} manufacturer={manufacturer} />} />
              <Route path='/catalog/cosmetics/:itemID' element={<Product setBasket={setBasket} />} />
              <Route path='/basket' element={<Basket basket={basket} setBasket={setBasket}/>}/>
              <Route path='/admin'  element={<Admin />} />
            </Routes>
        </div>
      <Footer/>
    </div>
  )
}
