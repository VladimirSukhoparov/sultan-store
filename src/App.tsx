import React, { useState } from "react";
import { Route, Routes } from "react-router-dom";

import data from "./data.json";

import { Header } from "./components/Header";
import { Footer } from "./components/Footer";
import { Breadcrumb } from "./components/Breadcrumb";
import { Home } from "./components/Home";
import { Product } from "./components/Product";
import { Catalog } from "./components/Catalog";
import { Cosmetics } from "./components/Cosmetics";
import { Basket } from "./components/Basket";
import { Admin } from "./components/Admin";

import "./App.scss";

export const App = () => {
  const [basket, setBasket] = useState(
    JSON.parse(localStorage.getItem("basket")) || []
  );
  const [products, setProducts] = useState(data);

  const manufacturer = Object.values(
    data.reduce((arr, e) => {
      let manufacturer = e.manufacturer;
      if (!arr[manufacturer]) arr[manufacturer] = { manufacturer, count: 1 };
      else arr[manufacturer].count++;
      return arr;
    }, {})
  );

  return (
    <div className="app">
      <Header
        basket={basket}
        products={products}
        data={data}
        setProducts={setProducts}
      />
      <div className="content">
        <Breadcrumb />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="catalog" element={<Catalog />} />
          <Route
            path="catalog/cosmetics"
            element={
              <Cosmetics
                data={products}
                setBasket={setBasket}
                manufacturer={manufacturer}
              />
            }
          />
          <Route
            path="catalog/cosmetics/:itemID"
            element={<Product setBasket={setBasket} />}
          />
          <Route
            path="basket"
            element={<Basket basket={basket} setBasket={setBasket} />}
          />
          <Route path="admin" element={<Admin />} />
        </Routes>
      </div>
      <Footer />
    </div>
  );
};
