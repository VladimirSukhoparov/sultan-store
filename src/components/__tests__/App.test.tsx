import { BrowserRouter } from "react-router-dom";
import { render, screen, cleanup } from "@testing-library/react";
import { App } from "../../App";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("should render app", () => {
  render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  const textElement = screen.getByText(/Упс! Что-то пошло не так.../i);
  expect(textElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  const app = renderer
    .create(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    )
    .toJSON();
  expect(app).toMatchSnapshot();
});
