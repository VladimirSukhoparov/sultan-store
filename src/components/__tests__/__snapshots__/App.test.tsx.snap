// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="app"
>
  <div
    className="header"
    data-testid="header"
  >
    <div
      className="header__container"
    >
      <div
        className="header__link"
      >
        <div
          className="header__box"
        >
          <img
            alt="geo"
            className="header__geo"
            src="geo.png"
          />
          <div>
            <p
              className="up__adress"
            >
              г. Кокчетав, ул. Ж. Ташенова 129Б
            </p>
            <p
              className="sub__adress"
            >
              (Рынок Восточный)
            </p>
          </div>
        </div>
        <div
          className="header__box"
        >
          <img
            alt="geo"
            className="header__mail"
            src="mail.png"
          />
          <div>
            <p
              className="up__adress"
            >
              opt.sultan@mail.ru
            </p>
            <p
              className="sub__adress"
            >
              На связи в любое время
            </p>
          </div>
        </div>
        <ul
          className="header__list"
        >
          <li
            className="header__item"
          >
            <a>
              О компании
            </a>
          </li>
          <li
            className="header__item"
          >
            <a>
              Доставка и оплата
            </a>
          </li>
          <li
            className="header__item"
          >
            <a>
              Возврат
            </a>
          </li>
          <li
            className="header__item"
          >
            <a>
              Контакты
            </a>
          </li>
        </ul>
      </div>
      <hr
        id="one__line"
      />
      <div
        className="header__contact"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            className="header__logo"
            src="logo.png"
          />
        </a>
        <a
          className="header__catalog"
          href="/catalog"
          onClick={[Function]}
        >
          Каталог
          <img
            alt="catalog"
            src="catalog.png"
          />
        </a>
        <div
          className="header__search"
        >
          <input
            className="header__input"
            onChange={[Function]}
            placeholder="Поиск..."
            type="search"
          />
          <button
            className="header__searchBtn"
          >
            <img
              alt="search"
              src="search.png"
            />
          </button>
        </div>
        <div
          className="header__tel"
        >
          <div
            className="header__call"
          >
            <p
              className="up__call"
            >
              +7 (777) 490-00-91
            </p>
            <p
              className="sub__call"
            >
              время работы: 9:00-20:00
            </p>
            <a>
              Заказать звонок
            </a>
          </div>
          <img
            alt="caller"
            className="header__caller"
            src="call.png"
          />
        </div>
        <a
          className="header__price"
        >
          Прайс-лист
          <img
            alt="download"
            src="download.png"
          />
        </a>
        <a
          className="header__basket"
          href="/basket"
          onClick={[Function]}
        >
          <img
            alt="basket"
            className="basket__img"
            src="basket.png"
          />
          <p
            className="basket__num"
          >
            0
          </p>
          <div
            className="basket__box"
          >
            <p
              className="basket__name"
            >
              Корзина
            </p>
            <p
              className="basket__price"
            >
              0
               
              ₸
            </p>
          </div>
        </a>
      </div>
      <hr
        id="two__line"
      />
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="breadcrumb"
    />
    <div
      className="home"
    >
      <h1
        className="home__title"
      >
        Упс! Что-то пошло не так...)))
      </h1>
      <a
        className="home__admin"
        href="/admin"
        onClick={[Function]}
      >
        Администрирование
      </a>
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="footer__container"
    >
      <div
        className="footer__mail"
      >
        <img
          alt="logo2"
          className="footer__logo"
          src="logo2.png"
        />
        <p
          className="footer__text"
        >
          Компания «Султан» — снабжаем розничные магазины товарами
          <br />
          "под ключ" в Кокчетаве и Акмолинской области
        </p>
        <p
          className="footer__signature"
        >
          Подпишись на скидки и акции
        </p>
        <div
          className="footer__subscribe"
        >
          <input
            className="footer__input"
            placeholder="Введите ваш E-mail"
            required={true}
            type="email"
          />
          <a
            className="footer__subscribeBtn"
          >
            <img
              alt="arrowButton"
              src="arrow.png"
            />
          </a>
        </div>
      </div>
      <div
        className="footer__menu"
      >
        <h2
          className="footer__title"
        >
          Меню сайта:
        </h2>
        <ul
          className="footer__list"
        >
          <li>
            <a
              className="footer__item"
            >
              О компании
            </a>
          </li>
          <li>
            <a
              className="footer__item"
            >
              Доставка и оплата
            </a>
          </li>
          <li>
            <a
              className="footer__item"
            >
              Возврат
            </a>
          </li>
          <li>
            <a
              className="footer__item"
            >
              Контакты
            </a>
          </li>
        </ul>
      </div>
      <div
        className="footer__category"
      >
        <h2
          className="footer__title"
        >
          Категории:
        </h2>
        <ul
          className="footer__list"
        >
          <li>
            <a
              className="footer__item"
            >
              Бытовая химия
            </a>
          </li>
          <li>
            <a
              className="footer__item"
            >
              Косметика и гигиена
            </a>
          </li>
          <li>
            <a
              className="footer__item"
            >
              Товары для дома
            </a>
          </li>
          <li>
            <a
              className="footer__item"
            >
              Товары для детей и мам
            </a>
          </li>
          <li>
            <a
              className="footer__item"
            >
              Посуда
            </a>
          </li>
        </ul>
      </div>
      <div
        className="footer__price"
      >
        <h2
          className="footer__title"
        >
          Скачать прайс-лист:
        </h2>
        <a
          className="footer__download"
        >
          Прайс-лист
          <img
            alt="download"
            src="download.png"
          />
        </a>
        <p
          className="footer__social"
        >
          Связь в мессенджерах:
        </p>
        <a>
          <img
            alt="whatsapp"
            className="footer__whatsapp"
            src="whatsapp.png"
          />
        </a>
        <a>
          <img
            alt="telegram"
            className="footer__telegram"
            src="telegram.png"
          />
        </a>
      </div>
      <div
        className="footer__contacts"
      >
        <h2
          className="footer__title"
        >
          Контакты:
        </h2>
        <div
          className="footer__call"
        >
          <p
            className="up__call"
          >
            +7 (777) 490-00-91
          </p>
          <p
            className="sub__call"
          >
            время работы: 9:00-20:00
          </p>
          <a>
            Заказать звонок
          </a>
        </div>
        <div
          className="footer__box"
        >
          <p
            className="up__adress"
          >
            opt.sultan@mail.ru
          </p>
          <p
            className="sub__adress"
          >
            На связи в любое время
          </p>
        </div>
        <img
          alt="visa"
          className="footer__visa"
          src="visa.png"
        />
        <img
          alt="master"
          className="footer__master"
          src="master.png"
        />
      </div>
    </div>
  </div>
</div>
`;
